Sources/NIOPosix/ThreadPosix.swift

```diff
@@ -74,8 +74,27 @@ private func sysPthread_create(
 
 typealias ThreadOpsSystem = ThreadOpsPosix
 
+struct PthreadWrapper: @unchecked Sendable {
```

Comment:
Nit: Can't we use our existing `UnsafeTransfer` for this?

Reply:
We could, but in this case I wanted a special-case wrapper that we can see, and aim to remove when it's no longer needed.